src/color.rs

```diff
@@ -420,6 +420,153 @@ impl Color {
         self.alpha = alpha;
         self
     }
+
+    /// Mix this color with the other given color, with the given weighting .
+    /// 
+    /// The weight determines the amount of the first color that will be used
+    /// in mixing as a percentage. So 50% means use each color equally, while 
+    /// 25% indicates to use one-quarter of this color, and seventy-five percent 
+    /// of the second.
+    /// 
+    /// The alpha channel of both colors are also combined in this way.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 40);
+    /// assert_eq!(mixed, Color::rgba(72., 44., 163., 0.88));
+    /// ```
+    /// 
+    /// Passing a value for `weight` that is not between 0 and 100 will result
+    /// in `Err(String)` being returned.
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// // This will panic
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 101);
+    /// ```
+    pub fn mix(&self, with: &Color, weight: u32) -> Self {
+        if weight > 100 {
+            assert!(weight <= 100, "{} is not a valid value for weight. Must be within 0-100 inclusive", weight);
+        }
+        // This algorighm cribbed from Sass (http://sass-lang.com/documentation/Sass/Script/Functions.html#mix-instance_method)
+        // with some modifications to use floating point facilities provided by Rust
+        let p = weight as f64 / 100.0;
+        let w = p.mul_add(2., -1.);
+        let a = self.alpha - with.alpha;
+
+        let w1 = if w * a == -1. {
+            (w + 1.) / 2.
+        } else {
+            ((w + a) / a.mul_add(w, 1.) + 1.) / 2.
+        };
+
+        let w2 = 1. - w1;
+
+        let r_mod = self.red.mul_add(w1, with.red.mul_add(w2, 0.)).round();
+        let g_mod = self.green.mul_add(w1, with.green.mul_add(w2, 0.)).round();
+        let b_mod = self.blue.mul_add(w1, with.blue.mul_add(w2, 0.)).round();
+        let a_mod = self.alpha * p + with.alpha * (1. - p);
+        Color::rgba(r_mod, g_mod, b_mod, a_mod)
+    }
+
+    /// Retrieve the hue for this `Color`. The returned value is in degrees 
+    /// between 0° and 360° that represents its position on the color wheel. 
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c: Color = "blue".into();
+    /// assert_eq!(201.0, c.hue());
+    /// ```
+    pub fn hue(&self) -> f64 {
+        self.to_hsl().0
```

Comment:
I really like your reuse of `to_hsl` here and below. I don't think that separating out the calculations for just getting the hue, saturation, etc. would be worth the effort, so I appreciate that you chose the simplest way to implement these methods with as little duplication as possible. :)